### PR TITLE
Remove chosen

### PR DESCRIPTION
**Pull request by jimmynotjim**
Removes the Chosen.js plugin from the code base in favor of our new multi-select, includes fixes for improper option values from @kurtw 

## Additions

- WordPress processors for proper option values

## Removals

- Removes Chosen library and it's initialization and styles

## Changes

- Updates common router to initialize the multi-select script on any select with the `multi` attribute
- Fixes the values for author and tag options to remove special characters

## Testing

- checkout branch and run `python cfgov/manage.py sheer_index`
- run `gulp build`
- navigate to `/blog`, `/newsroom`, and `/activity-log` and check out the two multi-select options

## Review

- @schaferjh 
- @duelj 
- @sebworks 
- @KimberlyMunoz 
- @anselmbradford 

## Screenshots

screen shot 2016-03-11 at 12 58 21 pm

## Todos

- There's a bug in the multi-select where selected options aren't displayed after form submission (though the filters are applied). Bugfix for that in a forthcoming PR.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/macros/filters.html [hunk 1]
```diff
@@ -178,11 +178,13 @@
...
 
-<form class="{{ options.additional_classes }}"
-      method="{{ options.form_method }}"
-      {{ "action=" ~ options.action if options.action != '/' }}
+<form class="{{ options.additional_classes or '' }}"
+      method="{{ options.form_method or 'get' }}"
+      {{ "action=" ~ options.action if options.action != '/' else '' }}
       >
-    <input type="hidden" name="csrfmiddlewaretoken" value="{{ csrf_token }}">
+    {% if options.addition_classes and options.addition_classes == 'post' %}
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/macros/filters.html]: Seems like this should be `{% if options.form_method and options.form_method == 'post' %}`

**Comment by sebworks** [comment]
happy

**Comment by sebworks** [comment]
This PR allows us to get rid of  `date-range-formatter.js` as well.

**Comment by sebworks** [comment]
Should the `Date Range` be wrapping to the next line?

File: cfgov/jinja2/v1/_includes/macros/filters.html (hunk 1 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/macros/filters.html]: Can we remove `options.addition_classes` also? Is that used anywhere?

**Comment by KimberlyMunoz** [comment]
I'm having trouble tabbing past the multiselect. I can't do it, it doesn't even make me go through our many, many options, it just stops at the multiselect.

**Comment by KimberlyMunoz** [comment]
I found the steps to replicate. It doesn't happen when you hold down the select button or tab really quickly, because it's a problem with transitions.

1. Tab to the filter box
2. Tab to the multiselect, leaving at least a beat between each tab click.
3. Your focus moves to the first checkbox of the multiselect as the box closes and aria-hidden toggles to true
4. Your focus is lost and you can't tab further. Future tab clicks only bring you silence. 

It doesn't look like anything caused by this PR, but it should get fixed before we get live since tabbing through it breaks your ability to hear the rest of the page on a screenreader.

**Comment by jimmynotjim** [comment]
@KimberlyMunoz That's a recorded issue from the original PR for the script, I'm going to update the select to use the same updates Ans made to the nav to avoid the tabbing issue.

**Comment by jimmynotjim** [comment]
@sebworks whoa, good catch, I didn't even notice that since I was only looking at the select. I think I remember this issue coming up in the wagtail filters, so I'll see what we did there and submit a PR for it separately.

**Comment by sebworks** [comment]
:+1:

**Comment by schaferjh** [comment]
One small step for man...one huge leap for accessibility!

File: cfgov/jinja2/v1/_includes/macros/filters.html (hunk 1 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/macros/filters.html]: @kurtw, a little help? This is from your commit, not sure why it changed.

Comment (kurtw) [inline cfgov/jinja2/v1/_includes/macros/filters.html]: whoops i meant to write `{% if options.form_method and options.form_method == 'post' %}`. Sorry for the confusion

Comment (kurtw) [inline cfgov/jinja2/v1/_includes/macros/filters.html]: We want the filter to post a CSRF token if the request method is POST but get requests shouldn't have it in there.

**Comment by kurtw** [comment]
Fixed the issue that checked the `options.optional_classes` instead of `options.form_method`

File: cfgov/unprocessed/js/routes/browse-filterable/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+var SecondaryNavigation = require( '../../organisms/SecondaryNavigation' );
+var Notification = require( '../../molecules/Notification' );
+var FilterableListControls = require( '../../organisms/FilterableListControls' );
+
+var secondaryNavigation = new SecondaryNavigation( document.body ).init();
+
+var notifications = document.querySelectorAll( '.m-notification' );
+var notification;
+for ( var i = 0, len = notifications.length; i < len; i++ ) {
+	notification = new Notification( notifications[i] ).init();
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/routes/browse-filterable/index.js]: Trivial: tab indent

**Comment by anselmbradford** [comment]
:+1: LGTM from a code perspective, I can't run it locally because indexing is erroring.

**Comment by KimberlyMunoz** [comment]
LGTM. 

Please make sure there's a JIRA issue for hiding the dropdown without killing the focus. Right now the page is less accessible than it was with chosen and it'd be good to have the issue to make sure it doesn't fall through the cracks. :no_good:

**Comment by jimmynotjim** [comment]
@KimberlyMunoz it's Jira#1008, set up for next sprint.

File: cfgov/unprocessed/js/routes/browse-filterable/index.js (hunk 2 above)
Comment (jimmynotjim) [inline cfgov/unprocessed/js/routes/browse-filterable/index.js]: This seems to be a Github error, because it's correct locally.

**Comment by KimberlyMunoz** [comment]
> It's Jira#1008, set up for next sprint.

Great! :+1:

**Pull request merged**